src/get-started/first-java-service.md

```diff
@@ -0,0 +1,615 @@
+
+#### Find Vehicle Service Operation
+
+First, we need to add a query operation to the Service:
+
+<!-- FIXME: Replace lines: selector with inside_block:ci-find-vehicle when
+the bug with braces is resolved: ECR-4318 -->
+<!--codeinclude-->
+[MyService.findVehicle](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+lines:138-146
+<!--/codeinclude-->
+
+Although this query method will be invoked by our code, hence the signature
+we use may be arbitrary, the signature is similar to the transaction methods:
+it takes the operation arguments and the context (here: `String` ID
+and `BlockchainData` context).
+
+#### API Controller
+
+<!-- As Exonum does not provide a means to invoke such read operations,
+we will
+-->
+
+Next, we will add a class implementing the REST API of the service.
+It will expose our "find vehicle" operation as a `GET` method.
+
+Create a new `ApiController` class:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/ApiController.java)
+block:ApiController
+<!--/codeinclude-->
+
+The `ApiController` needs a `MyService` object to query data; and
+a `Node` object to obtain the needed context: `BlockchainData`.
+
+!!! note
+    We encode the response in Protocol Buffers binary format for brevity.
+    The controller may encode it in any suitable format (e.g., JSON).
+
+Finally, connect the controller to the service. `MyService` already has
+an empty `createPublicApiHandlers` method, modify it to have:
+
+<!--codeinclude-->
+[MyService.createPublicApiHandlers](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-createPublicApiHandlers
+<!--/codeinclude-->
+
+!!! success
+    That's it with the service implementation! Package the _service artifact_
+    and run the integration tests:
+
+    ```shell
+    mvn verify
+    ```
+
+    and proceed to the next section, where we will test its operation.
+
+## Test Network
+
+Let's now launch a test network in which we can see our service operation.
+Our project already has a script launching a test network with a single
+validator node: `start-testnet.sh`.
+
+Run the script:
+
+```shell
+chmod 744 start-testnet.sh # Allow the script execution, needed once
+./start-testnet.sh
+```
+
+When you see messages like the following, the network is active:
+
+<!-- markdownlint-disable line-length -->
+```text
+[2020-03-05T10:36:24Z INFO  exonum_node::consensus] COMMIT ====== height=4, proposer=0, round=1, committed=0, pool=0, hash=43ac20f8b...
+```
+<!-- markdownlint-enable line-length -->
+
+Open a separate shell session and check the active services:
+
+```shell
+# You may pipe the response into `jq` to pretty-print it, if you have it
+# installed:
+curl -s http://127.0.0.1:3000/api/system/v1/services # | jq
```

Comment:
FYI: The URL for this endpoint has changed in post-RC Exonum. The endpoint is now available on `/api/services/supervisor/services`.

Reply:
Thanks, ideally, we shall include the _tested_ scripts too :-) 

Added https://jira.bf.local/browse/ECR-4324

Reply:
The current snippet works with 0.10.0-SNAPSHOT :-)